### PR TITLE
#167733660 Implement functionality for adding comments on Travel requests

### PR DESCRIPTION
**Pull request by daylay92**
#### What does this PR do?
Enable users to be able to comment on Travel requests.

#### Description of Task to be completed?
- Create the database models, migrations and establish appropriate relationships between tables.
- Create route, service method and controller method that enables a user to add a comment.
- Implement middleware method for validating input parameters
- Add the authorization and authentication middlewares.
- Write unit tests for this feature
- Document the new endpoint on swagger
#### How should this be manually tested?
- Clone the repo and cd into the apps working directory
- Checkout to the **ft-travel-request-comment-167733660** branch
- Create a .env file and specify the variables described in the .env.sample inside this file.
- Run `npm run migrate:reset` && `npm run migrate` && `npm run seed` to create tables in your 
 database and seed the database tables with the necessary app configs.
- Run docker-compose up --build to build the docker image and spin up all the necessary services
- As an alternative to running the app on docker, you can run `npm run start:dev`
- Launch Postman/insomnia and signup/login as a new company user. Refer to the swagger document.
- Make a travel request using the appropriate endpoint as documented on the swagger documentation.
- Make a POST request to `localhost:3000/api/comment` route using the fields described in the swagger document of this project.
- The appropriate data look's like this: 
```
           {
  "requestId": 1,
  "message": "What's happening with your trip?"
}
```
-  A successful request results in a JSON response with a status code of 201.
- To run the unit tests, run `npm run test`.
#### Any background context you want to provide?
This application is bootstrapped using docker as described above, however, the app can still be run locally without docker using the `npm run start:dev` 

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/167733660

### CODE REVIEW

**Review by Baystef — COMMENTED** [review]
Great job man, just complete your rebase and I'll approve

File: src/controllers/commentController.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+   * Creates a new comment on a travel request.
+   *
+   * @static
+   * @param {Request} req - The request from the endpoint.
+   * @param {Response} res - The response returned by the method.
+   * @returns { JSON } A JSON response containing details of the newly created comment.
+   * @memberof CommentController
+   */
+  static async create(req, res) {
+    try {
```
Comment (dchima) [inline src/controllers/commentController.js:24]: Maybe rename this `createComment` to reduce ambiguity when called in other files.
It doesn't matter if it's the same name as the service function.

**Review by dchima — APPROVED** [review]
All tests pass on local, code passes in my book. Well done!

**Review by Baystef — APPROVED** [review]

File: src/controllers/commentController.js (hunk 1 above)
Comment (daylay92) [reply src/controllers/commentController.js:24]: Thanks @chima. I would update the name as suggested

**Review by amarachukwu-agbo — COMMENTED** [review]
LGTM

**Pull request merged**